### PR TITLE
fix pytorch 2.0 optimizer no inf check error

### PR DESCRIPTION
**Pull request by ksnzh**
fix #1883 #1613.

The reason why there's no inf check is because proposal network did not compute gradient at some iteration step. And there is no need to do optimizer.step for proposal network's parameter.

### CODE REVIEW

**Comment by ksnzh** [comment]
It's compatibale with ptorch 1.x and 2. Here's my experient: https://wandb.ai/ksnzh/nerfstudio-public/runs/b897dfzk and https://wandb.ai/ksnzh/nerfstudio-public/runs/zqdvp8vu
